gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -237,6 +233,16 @@ public void run() {
         LOG.error("Failed to close all open resources", t);
       }
 
+      for (Map.Entry<Optional<Fork>, Optional<Future<?>>> forkAndFuture : this.forks.entrySet()) {
```

Comment:
I can but its slightly more complicated than it seems. The current PR preserves the current behavior with regards to isolation of failures between `Fork`s, but I don't think it is the ideal behavior.

For now it may be best to stick to the current approach and add better isolation semantics if there is need. The problem is a `Task` doesn't really support re-trying from a known good location, its all or nothing:
* We would have to carry over some state from the failed `Task` to the new `Task`
* We also need to consider which files get deleted if only some `Fork`s fail